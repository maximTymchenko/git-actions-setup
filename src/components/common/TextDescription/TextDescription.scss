@import '../../../styles/vars';

.text-description {
    font-family: 'OktaNeue-Regular';
    max-width: 680px;
    font-size: 18px;
    line-height: 1.53;
    color: $text__body;

    @media screen and (max-width: $small__desktop) {
        max-width: 100%;
    }

    @media screen and (max-width: $mobile) {
        font-size: 16px;
        max-width: 400px;
    }

    @media screen and (max-width: $small__mobile) {
        max-width: 350px;
    }

    &--mb {
        margin-bottom: 20px;
    }

    &--subparagraph {
        margin-left: 25px;
        margin-bottom: 20px;
    }

    &--large {
        font-size: 22px;

        @media screen and (max-width: $mobile) {
            font-size: 19px;
            // width: 330px;
            padding-right: 10px;
        }
    }

    &--small {
        font-size: 15px;

        @media screen and (max-width: $mobile) {
            // width: 330px;
            font-size: 15px;
            padding-right: 10px;
        }
    }

    &--bold {
        cursor: pointer;
        font-family: 'OktaNeue-Medium';
        font-size: 19px;
        font-weight: 500;
        line-height: 1.53;
        color: $secondary;
        margin-left: 5px;
        margin-right: 5px;
    }

    &--dark {
        font-size: 15px;
        color: $text;
    }
}
