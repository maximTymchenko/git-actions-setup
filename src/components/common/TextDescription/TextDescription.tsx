import React, { Children, FC } from 'react';
import './TextDescription.scss';

interface ITextDescription {
    description: string;
    type: string;
    children: string;
}

const TextDescription: FC = (props: ITextDescription) => {
    return (
        <p
            data-testid="text-description"
            className={`text-description text-description--${props.type}`}
        >
            {props.children}
        </p>
    );
};

export default TextDescription;
