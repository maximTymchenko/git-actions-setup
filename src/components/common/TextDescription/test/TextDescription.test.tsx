import React from 'react'
import TextDescription from '../TextDescription'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <TextDescription description="test"></TextDescription>
  )
  expect(getByTestId('text-description')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(<TextDescription description="test"></TextDescription>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
