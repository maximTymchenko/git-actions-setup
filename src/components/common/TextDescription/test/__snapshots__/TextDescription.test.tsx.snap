// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<p
  className="text-description text-description--undefined"
  data-testid="text-description"
/>
`;
